src/java/org/apache/cassandra/tools/JMXTool.java

```diff
@@ -393,9 +393,16 @@ private static final class CustomConstructor extends Constructor
         {
             private static final String ROOT = "__root__";
             private static final TypeDescription INFO_TYPE = new TypeDescription(Info.class);
+            private static final org.yaml.snakeyaml.LoaderOptions LOADER_CONFIG = new org.yaml.snakeyaml.LoaderOptions();
+            {
+                // Set the max yaml file size to 30 mb.
+                LOADER_CONFIG.setCodePointLimit(31_457_280);
```

Comment:
It looks like some tests are failing with
```
Exception in thread "main" org.yaml.snakeyaml.error.YAMLException: The incoming YAML document exceeds the limit: 3145728 code points.
```

only the value used is an order of magnitude too small?

Reply:
Yeah. I thought this had fixed that. Guess not. I'll have to dig around some more it seems.

snakeyaml added a default max of 3 MB in this new release to protect from bad input crashing people.  It is still saying that default for some reason not the raised value from here :(.

Reply:
Ok figured it out.  The "Yaml" object also needs the config passed to it, it saves its own copy rather than getting it out of the "Constructor" object.